docs: rework setup instructions (#27959)
sourcegraph/sourcegraph: doc/local-dev-env/alternative-set-up-for-databases.md

# Alternative set up for databases

> The sourcegraph codebase requires a few databases to be installed in order to run locally. We recommend and support a default approach that is suited for everyone, regardless of their familiarity with the topic. 

Reviewer: ```suggestion
Sourcegraph requires databases to be accessible to run locally. We recommend and support a default approach that is suited for everyone, regardless of their familiarity with the topic. 
```

> It's totally possible to use alternative ways to install databases if you need to: `sg` checks will still perform correctly, but you're on your own for the installation process: see the [alternative instructions section](#alternative-instructions).

Reviewer: ```suggestion
But it's still possible to use alternative ways to install databases if you want to and know what you're doing: `sg checks` will still perform correctly, but you're on your own for the installation process: see the [alternative instructions section](#alternative-instructions).
```

> ## MacOs

Reviewer: ```suggestion
## macOS
```

### Using packaged applications

Requirements: 

- The `github.com/sourcegraph/sourcegraph` repository cloned in a folder of your choice.

> - [`Homebrew`](https://brew.shell) is installed.

Reviewer: I don't think this needs to be here? It's general and applies to a lot of things, but it _doesn't_ apply to `Postgres.app` and `Redis.app`.

Instructions:

- Install _Postgres.app_:
  1. Open your browser and navigate to this page: [https://postgresapp.com](https://postgresapp.com)
  1. Follow the installation instructions.
- Install _Redis.app_:
  1. Open your browser and navigate to this page: [https://jpadilla.github.io/redisapp/](https://jpadilla.github.io/redisapp/)
  1. Click on the Download button and install the app in the archive.
  1. Open a terminal and type: `sudo mkdir -p /etc/paths.d && echo /Applications/Redis.app/Contents/Resources/Vendor/redis/bin | sudo tee /etc/paths.d/redisapp`

> ### Using Homebrew

Reviewer: ```suggestion
### Using Homebrew (recommended!)
```

Reviewer: I think we should make this the recommended way since it's the easiest to automatically run with `sg setup` and also the easiest from a "get `psql` into `$PATH`" perspective
